test/unit/black-button_test.js

```diff
@@ -84,9 +84,152 @@ describe('Black button testing', function() {
         });
     });
 
-    describe('When the creation in the CB returns an error: ', function() {
-        it('should return an explanation of the kind of error to the device');
-    });
+    function generateAsynchOrionErrorCase(action, payload, cbRequest, errorFile, queryRequest, queryError) {
+        return function() {
+            var options = {
+                    url: 'http://localhost:' + config.thinkingThings.port + config.thinkingThings.root + '/Receive',
+                    method: 'POST',
+                    form: {
+                        cadena: payload
+                    }
+                },
+                originalGenerateInternalId;
+
+            beforeEach(function(done) {
+                var mocks = [];
+
+                config.ngsi.plainFormat = true;
+
+                originalGenerateInternalId = idGenerator.generateInternalId;
+                idGenerator.generateInternalId = mockedGenerateInternalId;
+
+                if (queryRequest && queryError) {
+                    mocks.push(utils.prepareMocks(
+                        './test/unit/contextRequests/' + queryRequest,
+                        './test/unit/contextResponses/' + queryError,
+                        '/v1/queryContext'));
+                }
+
+                mocks.push(utils.prepareMocks(
+                    './test/unit/contextRequests/' + cbRequest,
+                    './test/unit/contextResponses/' + errorFile,
+                    '/v1/updateContext'
+                    ));
+
+                async.series(mocks, done);
+            });
+
+            afterEach(function() {
+                config.ngsi.plainFormat = false;
+                idGenerator.generateInternalId = originalGenerateInternalId;
+            });
+
+            it('should return an explanation of the kind of error to the device', function(done) {
+                request(options, function(error, result, body) {
+                    should.not.exist(error);
+                    result.statusCode.should.equal(200);
+                    body.should.equal('#STACK1#0,BT,' + action + ',0,0:500,rgb-66CC00;t-2,0$');
```

Comment:
Since the code `rgb-66CC00;t-2` is part of the corresponding `.json` file used in this test case, what about including it as `require('xxx.json').contextResponses.statusCode.reasonPhrase` instead of as hard-coded text in the test case assertions (of course, the `require()` could be done in the 'beforeEach()' function (if not already done) or any place you consider appropriate... :)